src/IssueViz.Security/OpenInIDE/Api/OpenInIDEFailureInfoBar.cs

```diff
@@ -0,0 +1,99 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.InfoBar;
+using SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.OpenInIDE.Api
+{
+    internal interface IOpenInIDEFailureInfoBar
+    {
+        void Show();
+
+        void Clear();
+    }
+
+    [Export(typeof(IOpenInIDEFailureInfoBar))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class OpenInIDEFailureInfoBar : IOpenInIDEFailureInfoBar, IDisposable
```

Comment:
The new class is pretty straightforward too. It's almost exactly the same as it was in the validator class. The only difference is the addition of the `Clear` method so callers can remove old bars if necessary (e.g. if we receive a new "open" request that we can process successfully).